CMakeLists.txt

```diff
@@ -400,10 +398,8 @@ if(WITH_PROMETHEUS)
   endif()
 endif()
 
-if(WITH_ABSEIL)
-  if(NOT TARGET absl::strings)
-    find_package(absl CONFIG REQUIRED)
-  endif()
+if(WITH_OTLP_GRPC)
+  find_package(absl CONFIG REQUIRED)
```

Comment:
Why is absl only required for `WITH_OTLP_GRPC`, not for all cases?

Reply:
otel-cpp's API do not depend abseil-cpp's headers now, and there is no conlict paths between internal version and external version  of abseil-cpp.

There are some components use both protobuf and it may include abseil-cpp's headers which have the same path as the internal version. That's why we need abseil-cpp when we use a new version of protobuf.

The protobuf's package file will call `find_package(absl CONFIG REQUIRED)` and import it when need. But some versions of gRPC do not import it, so we only need to import it when using gRPC.